### PR TITLE
[NO GBP] Debug chem synthesizer works & related code cleanup

### LINKED ISSUE #79610: Debug Chem Synthesizer broken
## Testmerges:

None.

## Reproduction:

Attempt to use: code/modules/reagents/chemistry/machinery/chem_synthesizer.dm.
It no longer can add reagents to beakers, only create them. This was probably an oversight.

Broken by https://github.com/tgstation/tgstation/pull/79395#issue-1968696051
<details>
  <summary>Before</summary> 

https://github.com/tgstation/tgstation/assets/106436013/3a1ad7eb-b015-41bc-83ab-a4aa04960769

</details>

<details>
  <summary>After</summary> 

https://github.com/tgstation/tgstation/assets/106436013/3fdd8da0-b336-47d5-a4d1-08f0cf600571

</details>

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
1. Fixes #79610
    
    It works and the UI has also been updated to match that of chem dispenser and it displays an input list for adding reagents now
    
    Screenshot (355)

2. Created a common component for chem dispenser, portable chem dispenser & debug chem synthesizer to share in the UI code.

3. Moved portable chem mixer to modules/reagents/machinery folder to group them under the chemistry category

## Changelog
:cl:
fix: debug chem synthesizer works again. cleaned up chem dispenser, portable chem dispenser & debug chem synthesizer ui code
qol: ui for displaying beaker reagents for debug chem synthesizer has been improved. Now displays input list for adding reagents
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
Its been a week hmmm. Time to phone a friend. @Time-Green can you help me out :D ?

**Review by Time-Green — APPROVED** [review]
DM code is fine, but I can't review UI. @jlsnow301 can you do a UI review?

File: tgui/packages/tgui/interfaces/common/BeakerDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+export type Beaker = {
+  maxVolume: number;
+  transferAmounts: number[];
+  pH: number;
+  currentVolume: number;
+  contents: BeakerReagent[];
+};
+
+export type BeakerProps = {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/common/BeakerDisplay.tsx:18]: are we using these types elsewhere? if not, prefer not exporting them

Comment (SyncIt21) [reply tgui/packages/tgui/interfaces/common/BeakerDisplay.tsx:18]: Yes we are. I did say this in the PR body 

> Created a common component for chem dispenser, portable chem dispenser & debug chem synthesizer to share in the UI code.

This is that component

Comment (jlsnow301) [reply tgui/packages/tgui/interfaces/common/BeakerDisplay.tsx:18]: You're using the component and Beaker sure- I don't see any imports of BeakerReagent or BeakerProps

Comment (SyncIt21) [reply tgui/packages/tgui/interfaces/common/BeakerDisplay.tsx:18]: Ah my bad it was like 1pm yesterday so I wasn't thinking straight. 
Yup I'll remove the export

**Pull request merged**